### ISSUE TITLE
SpeechRecognition: Packaging Error of localized SGRS-Folders

### ISSUE THREAD

**Issue by DamirLisak** [Microsoft/Windows-universal-samples#536]
In my VS-Project I have included two XML-Files (for Speech-Recognition) and set the Properties to Content according to the example in the SpeechRecognitionAndSynthesis-Project:

```
    <Content Include="SRGS\de-DE\SRGSMatch.xml">
      <SubType>Designer</SubType>
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
    <Content Include="SRGS\en-US\SRGSMatch.xml">
      <SubType>Designer</SubType>
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
```

When I run my App in Debug- or Release-Mode I can Access both files.
But after deploying the App-Package to the store (sucessful submission) the users who download my App don't have the "SRGS\de-DE"-Folder in their Application-Directory. Only the "SRGS\en-US"-Folder is created.
After this I have searched the file "SRGSMatch.xml" in the Build-Directory of the package and I found out, that the "SRGS\de-DE\SRGSMatch.xml"-File doesn't appear in the filemap.map.txt. This is the search-Result:

```
C:\MyProject.uwp\obj\x86\Release\Upload\filemap.language-de.map.txt(7):"C:\MyProject.uwp\obj\x86\Release\PackageUploadLayout\SRGS\de-DE\SRGSMatch.xml"			"SRGS\de-DE\SRGSMatch.xml"
  C:\MyProject.uwp\obj\x86\Release\Upload\filemap.map.txt(150):"C:\MyProject.uwp\obj\x86\Release\PackageUploadLayout\SRGS\en-US\SRGSMatch.xml"			"SRGS\en-US\SRGSMatch.xml"
  C:\MyProject.uwp\obj\x86\Release\Upload\main.map.txt(152):"C:\MyProject.uwp\obj\x86\Release\PackageUploadLayout\SRGS\en-US\SRGSMatch.xml"			"SRGS\en-US\SRGSMatch.xml"
  C:\MyProject.uwp\obj\x86\Release\Upload\split.language-de.map.txt(7):"C:\MyProject.uwp\obj\x86\Release\PackageUploadLayout\SRGS\de-DE\SRGSMatch.xml"			"SRGS\de-DE\SRGSMatch.xml"
  C:\MyProject.uwp\obj\x86\Release\Upload\split.map.txt(9):"C:\MyProject.uwp\obj\x86\Release\PackageUploadLayout\SRGS\en-US\SRGSMatch.xml"			"SRGS\en-US\SRGSMatch.xml

```

The Visual-Studio-Packaging-Tool has a problem when the localized SGRS-Files are organized in the "BCP-47 language tag"-Manner. 
I wrote this issue to the Windows-Store-Support, but the didn't react.

**Comment by anpilley** [comment]
Just to clarify, what you're seeing is that en-US customers only see the en-US SRGS grammar files? And de-DE customers can only see the de-DE grammar files?

I believe this is generally expected, based on how localization works. When you generate a package bundle, the packager creates a series of localized appx's, like "SpeechAndTTS_1.0.0.0_language-de". Each one will only contain it's own language files. (when I extract out the appx, then extract out the content of the appx, it only contains SRGS\de-DE\SRGSColors.xml)

It probably should be clearer in the sample, but if your intention is to allow de-DE language usage for en-US customers, this may require some alternate mechanisms to enable (ie, you may need to use some kind of lookup system). The primary concern is that those en-US customers will still need to have installed the de-DE language packs for de-DE to be available to them. 

Altering how localized assets are packaged and distributed by the store is beyond the scope of the Speech and Recognition APIs, although I'll take that under advisement if updates to the sample are warranted.

**Comment by DamirLisak** [comment]
@anpilley : No, the de-DE customers can't see the de-DE grammar files.
I try to load the file with:
`grammarContentFile = await StorageFile.GetFileFromApplicationUriAsync(new System.Uri(String.Format("ms-appx:///SRGS/{0}/SRGSMatch.xml", sgrsLanguage)));`
German customers who have installed my App over the store get's a Exception, that the file coudn't be found. When I open the installation directory of my app there is no "de-DE"-Folder.

When I unzip the App-Packacke (...bundle.appxupload) I find a "_language-de.appx" inside. it contains the SRGS-Folder + follwing files:
```
AppxBlockMap.xml
AppxManifest.xml
AppxSignature.p7x
resources.pri
[Content_Types].xml
<DIR>          SRGS
```
The SRGS-Folder contains the "de-DE"-Folder with the Grammar-File.

I think it has something to do with the packaging and the Installation of the App.
To solve the Problem I did a Workaround: Instead of creating directories for each language I created only Files with a language postfix. e.g.: ...de-DE.xml so that they are not placed in the language-dependend-Packages.
I load them with:
`grammarContentFile = await StorageFile.GetFileFromApplicationUriAsync(new System.Uri(String.Format("ms-appx:///SRGS/SRGSMatch-{0}.xml", sgrsLanguage)));`
